gems/aws-sigv4/lib/aws-sigv4/signer.rb

```diff
@@ -111,41 +111,42 @@ class Signer
       # @option options [Array<String>] :unsigned_headers ([]) A list of
       #   headers that should not be signed. This is useful when a proxy
       #   modifies headers, such as 'User-Agent', invalidating a signature.
+      #   All header names are downcased.
+      #
+      # @option options [Symbol, String] :signing_algorithm ('sigv4') The algorithm to use for signing.
+      #   This can either be `sigv4`, `sigv4a`, or `sigv4-s3express`.
       #
       # @option options [Boolean] :uri_escape_path (true) When `true`,
       #   the request URI path is uri-escaped as part of computing the canonical
       #   request string. This is required for every service, except Amazon S3,
       #   as of late 2016.
       #
-      # @option options [Boolean] :apply_checksum_header (true) When `true`,
-      #   the computed content checksum is returned in the hash of signature
-      #   headers. This is required for AWS Glacier, and optional for
-      #   every other AWS service as of late 2016.
-      #
-      # @option options [Symbol] :signing_algorithm (:sigv4) The
-      #   algorithm to use for signing.
+      # @option options [Boolean] :normalize_path (true) When `true`, the
+      #   uri paths will be normalized when building the canonical request.
       #
       # @option options [Boolean] :omit_session_token (false)
       #   (Supported only when `aws-crt` is available) If `true`,
       #   then security token is added to the final signing result,
       #   but is treated as "unsigned" and does not contribute
       #   to the authorization signature.
       #
-      # @option options [Boolean] :normalize_path (true) When `true`, the
-      #   uri paths will be normalized when building the canonical request.
+      # @option options [Boolean] :apply_checksum_header (true) When `true`,
+      #   the computed content checksum header (`x-amz-content-sha256`) is signed and returned
+      #   in the hash of signature headers. This is required for AWS Glacier, and optional
+      #   for every other AWS service as of late 2016.
       def initialize(options = {})
         @service = extract_service(options)
         @region = extract_region(options)
         @credentials_provider = extract_credentials_provider(options)
-        @unsigned_headers = Set.new((options.fetch(:unsigned_headers, [])).map(&:downcase))
+        @unsigned_headers = Set.new(options.fetch(:unsigned_headers, []).map(&:downcase))
         @unsigned_headers << 'authorization'
         @unsigned_headers << 'x-amzn-trace-id'
         @unsigned_headers << 'expect'
+        @signing_algorithm = options.fetch(:signing_algorithm, 'sigv4').to_s
```

Comment:
This used to be a symbol but I made it a string, and call to_s for compatibility. s3 express symbolized naming made this weird.